tools/e2etesting/steps/variables.yml

```diff
@@ -5,4 +5,8 @@ variables:
   AgentPool: Azure-IoT-Manufacturing
   IAILocalFilename: Microsoft.Azure.IIoT.Deployment.exe
   TestPath: $(System.DefaultWorkingDirectory)\e2e-tests\IIoTPlatform-E2E-Tests
-  Runtime: 'win-x64'
\ No newline at end of file
+  Runtime: 'win-x64'
+  ${{ if startsWith(variables['Build.SourceBranch'], 'refs/heads/') }}:
+    BranchName: $[ replace(variables['Build.SourceBranch'], 'refs/heads/', '') ]
+  ${{ if startsWith(variables['Build.SourceBranch'], 'refs/pull/') }}:
+    BranchName: $[ replace(variables['System.PullRequest.TargetBranch'], 'refs/heads/', '') ]
```

Comment:
Is this correct? checking SourceBranch to start with `/ref/pull` and then replace in target branch `refs/head/` with empty string without checking if traget branch startswith refs/head?

Reply:
As per the discussion, System.PullRequest.TargetBranch wont be needed to identify branchname for triggering end to end tests.